prometheus/metrics.go

```diff
@@ -346,45 +346,40 @@ func replaceInvalidCharacters(metricName string) string {
 	return invalidLabelCharRE.ReplaceAllString(metricName, "_")
 }
 
-func getNamespaceRequestRates(api v1.API, namespace string, ratesInterval string) (model.Vector, model.Vector, error) {
-	reporter := "source"
+func getAllRequestRates(api v1.API, namespace string, ratesInterval string) (model.Vector, model.Vector, error) {
+	outReporter := "source"
 	if config.Get().IstioNamespace == namespace {
-		reporter = "destination"
+		outReporter = "destination"
 	}
-
-	// traffic originating outside the namespace to destinations inside the namespace
```

Comment:
yeah but they were incorrect. It's not inside/outside namespace, it's inbound/outbound workload. But I will change the comment instead of removing it.

Reply:
(note that this part of the commit is not exactly new code, it's more or less reverting some incorrect changes that were done during istio-1 developments. I'm just "rollbacking" to the behaviour we had pre-istio-1, with just some adjustments regarding services vs. workloads)

Reply:
tbh I don't really like how this function is written & what it does, i'm gonna refactor it

Reply:
done: I'm going back to @jshaughn ideas for fetch namespace rates without duplicates ; the only thing is that it changes a bit the way responses are used by the caller. Better now?